enhancements/installer/openstack-supported-versions.md

```diff
@@ -0,0 +1,217 @@
+  is partially upgraded to a major version or if there is a Support
+  Exception granted to run that deprecated version of OpenStack.
+
+### Non-Goals
+
+This enhancement does not intend do block any OpenShift operations done
+outside of the installer if the OpenStack version is not supported.
+
+## Proposal
+
+In order to implement this feature fully, the following changes must be made
+in the openshift/installer:
+
+- Update gophercloud vendoring to (at least) [b56bf1](https://github.com/gophercloud/gophercloud/pull/2037/commits/5a1daf082451587459058cc539c53f2ae1aaaef3)
+  so we can benefit from the apiversions package.
+- Create a constant that will list all the supported versions
+  of Compute API based on the upstream manual.
+- Add a new validation against the OpenStack cloud to make sure that
+  the Compute API version is supported.
+- Create a configuration flag `ignoreVersionCheck` that when set to
+  false will send a warning if the OpenStack version isn't supported and
+  proceed anyway.
+
+### Design Details
+
+The OpenStack Compute service (also known as Nova) is always deployed
+in OpenStack clouds. Also, it is one of the services that bumps the
+microversions so regularly that is it possible to figure out
+what version of OpenStack is running based on the maximum version of
+Nova API that is available for a cloud.
+There's a contractual given here, which is: Nova will not bump the microversion
+of a stable release. There's also an assumption that Nova will always bump the
+microversion between stable releases (i.e. add or change the public API).
+This is a reasonable assumption which is unlikely to ever be broken.
+If it were ever not true we would still have an opportunity to look for
+alternatives.
+
+Therefore, it is proposed that we use that information as a reliable
+way of checking if the version of OpenStack that is running is
+actually supported.
+
+The Nova API versions that are supported will be maintained in
+the installer code and will follow the recommendations from
+the [upstream Nova API manual](https://docs.openstack.org/nova/latest/reference/api-microversion-history.html).
+
+There are multiple reasons where an administrator would want to ignore
+the OpenStack version check:
+- the administrators know what they are doing and understand that at this
+  point they aren't supported anymore.
+- the administrators have performed a major upgrade or Fast-Forward-Upgrade
+  of OpenStack and the operation is partially finished; however
+  they need to proceed to an OpenShift operation (e.g. scale up).
+  It it possible to run an OpenStack cloud with various versions of
+  APIs, as long as compatibility constraints are respected (out of
+  scope for this document).
+- the administrators have a Support Exception and an OpenStack upgrade
+  isn't possible at the moment of the operation.
+
+In that case, we would use a new configuration flag named `ignoreVersionCheck`.
```

Comment:
Upgrades are run using the `oc` tool, and not the Installer. How does the user set the flag for upgrades?

I see two scenarios here:
a. `ignoreVersionCheck` is persisted somewhere in a configmap. However, this has the consequence that (unless fancy logic) the property is set per-cluster, as opposed to per-operation.
b. a `ignoreVersionCheck` flag is added to `oc adm`.

Reply:
I think this enhancement initially targets the openshift/installer. I didn't think we would change `oc`'s behaviour. Do you think it would be worth it?
The initial use case is to prevent our customers to install OpenShift on top of a version of OpenStack that is not supported.

Reply:
@sferich888 What are your thoughts on the upgrade use case?

Reply:
We need to account for oc as well, and upgrades, however the upgade (oc) has a force option that we might be able to leverage here to meet the requirements of forcing an upgrade when such a check / block it hit. 

We should talk out what is best with @sttts or @mfojtik on this (or possibly @wking - I think you were involved in the initial upgrade/force discussions).